lib/facts/linux/scope6_interfaces.rb

```diff
@@ -10,15 +10,14 @@ def call_the_resolver
         resolved_facts = []
         scope6_interfaces = {}
         interfaces = Facter::Resolvers::NetworkingLinux.resolve(:interfaces)
-        primary = Facter::Resolvers::NetworkingLinux.resolve(:primary_interface)
+        primary_scope6 = Facter::Resolvers::NetworkingLinux.resolve(:scope6)
+
         interfaces&.each do |interface_name, info|
           scope6_interfaces["scope6_#{interface_name}"] = info[:scope6] if info[:scope6]
         end
```

Comment:
wouldn't
``resolved_facts << Facter::ResolvedFact.new("scope6_#{interface_name}", info[:scope6], :legacy) if info[:scope6]`` be more suitable here, to drop ``scope6_interfaces = {}`` and get rid of scope6_interfaces iteration?